src/js/components/Larivaar/util.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+
+const matrasThatAppearAtRightSideOfChar = "wIuUyYoOW"
+const halfCharThatAppearAtRightSideOfChar = "Í´R@˜®"
+const nasalSoundCaharacter = "NMµ"
+const sihari = "i";
+const allMatrasHalfCharAndNasalSoundChar = sihari + matrasThatAppearAtRightSideOfChar + halfCharThatAppearAtRightSideOfChar + nasalSoundCaharacter
+const unicodeMatras = "ਾਿੀੁੂੇੈੋੌੰਂ"
+
+export function fixLarivaar(word, unicode, larivaarAssistColor, indexVal) {
+    var breakSupportedCharList = [];
+    
+    var segmentedValArr = null;
+    if(unicode) {
+        segmentedValArr = fixLarivaarUnicode(word);
+    } else {
+        segmentedValArr = fixLarivaarGurmukhiFont(word);
+    }
+    
+    // Add each segment with wbr in a new span
+    for(let i=0;i<segmentedValArr.length;i++) {
+        if(segmentedValArr[i].indexOf("´") != -1) {
+        // handle space break for this special character
+        breakSupportedCharList.push(
+            <span key={i} className = "wordBreakSpecialChar">
+            {segmentedValArr[i]}<wbr />
+            </span>
+        )    
+        } else {
+            breakSupportedCharList.push(
+                <span key={indexVal+"."+i} style={{ color: indexVal % 2 === 1 ? larivaarAssistColor : '' }}>
+                    <span key={i}>
+                    {segmentedValArr[i]}<wbr />
+                    </span>
+                </span>
+            )
+        }
+    }    
```

Comment:
can you refactor this to use [Array.prototype.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)?

```jsx
const Characters = segmentedValArray
  .map(value => (
    value.includes("´")
      ?   <span key={i} className = "wordBreakSpecialChar">
            {segmentedValArr[i]}<wbr />
            </span>
  ))
  
```

Reply:
Done